artemis-cli/src/test/java/org/apache/activemq/artemis/cli/commands/CreateTest.java

```diff
@@ -0,0 +1,105 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.cli.commands;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.activemq.artemis.utils.XmlProvider;
+import org.apache.activemq.cli.test.CliTestBase;
+import org.apache.activemq.cli.test.TestActionContext;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.xml.sax.SAXException;
+
+@RunWith(Parameterized.class)
+public class CreateTest extends CliTestBase {
+
+   private final String testName;
+   private String httpHost;
+   private boolean relaxJolokia;
+
+   public CreateTest(String testName, String httpHost, boolean relaxJolokia) throws IOException {
+      this.testName = testName;
+      this.httpHost = httpHost;
+      this.relaxJolokia = relaxJolokia;
+   }
+
+   @Parameterized.Parameters(name = "{0}")
+   public static Collection<Object[]> testData() {
+      return Arrays.asList(new Object[][]{
+         {"Happy path + relaxJolokia", "sampledomain.com", true},
+         {"Happy path - relaxJolokia", "sampledomain.net", false},
+         {"Domain with dash + relaxJolokia", "sample-domain.co", true},
+         {"Domain with dash - relaxJolokia", "sample-domain.co.uk", false},
+         {"Domain with double dashes + relaxJolokia", "sample--domain.name", true},
+         {"Domain with double dashes - relaxJolokia", "sample--domain.biz", false},
+         {"Domain with leading dashes + relaxJolokia", "--sampledomain.company", true},
+         {"Domain with leading dashes - relaxJolokia", "--sampledomain.email", false},
+         {"Domain with trailing dashes + relaxJolokia", "sampledomain--.shop", true},
+         {"Domain with trailing dashes - relaxJolokia", "sampledomain--.java", false},
+      });
+   }
+
+   @Test
+   public void testWriteJolokiaAccessXmlCreatesValidXml() throws Exception {
+      TestActionContext context = new TestActionContext();
+      File testInstance = new File(temporaryFolder.getRoot(), "test-instance");
+
+      Create c = new Create();
+      c.setNoAutoTune(true);
+      c.setInstance(testInstance);
+      c.setHttpHost(httpHost);
+      c.setRelaxJolokia(relaxJolokia);
+      c.execute(context);
+
+      try {
+         isXmlValid(new File(testInstance, "etc/" + Create.ETC_JOLOKIA_ACCESS_XML));
+      } catch (Exception e) {
+         Assert.fail(testName + " should be valid, but " + e.toString());
```

Comment:
The `isXmlValid` method is catching the exceptions and converting them in a boolean but the returned boolean isn't checked.

You could check the returned boolean or remove the try/catch from the `isXmlValid` method.

I would remove the try/catch from the `isXmlValid` method to print the cause exception when it fails.

Reply:
You are correct.  I missed that when adding your suggestions.

I'll change this to `Assert.true(isXmlValid(new File(testInstance, "etc/" + Create.ETC_JOLOKIA_ACCESS_XML)));`, and remove the try/catch.